common/utilities/mdc.py

```diff
@@ -0,0 +1,21 @@
+import contextvars
+
+from comms.http_headers import HttpHeaders
+
+message_id: contextvars.ContextVar[str] = contextvars.ContextVar('message_id', default='')
+correlation_id: contextvars.ContextVar[str] = contextvars.ContextVar('correlation_id', default='')
+inbound_message_id: contextvars.ContextVar[str] = contextvars.ContextVar('inbound_message_id', default='')
+interaction_id: contextvars.ContextVar[str] = contextvars.ContextVar('interaction_id', default='')
+
+
+def build_tracking_headers():
+    headers = {}
+    if correlation_id.get():
+        headers[HttpHeaders.CORRELATION_ID] = correlation_id.get()
+    if message_id.get():
+        headers[HttpHeaders.MESSAGE_ID] = message_id.get()
+    if interaction_id.get():
+        headers[HttpHeaders.INTERACTION_ID] = interaction_id.get()
+    if inbound_message_id.get():
+        headers[HttpHeaders.INBOUND_MESSAGE_ID] = inbound_message_id.get()
+    return headers or None
```

Comment:
I'm not sure how to deal with that in python: is `None` better than empty set? I'd return just empty set as empty set resolves to false and is iterable (can use `in`/`not in` on it)

Reply:
Do you have test for case when `headers` is `None`?

Reply:
I used `None` because it's how it was used at https://github.com/nhsconnect/integration-adaptors/pull/186/files#diff-b1db7b05027c20d67989fe0d09d531d8L59

Reply:
Ok we can leave it, I'm not that familiar with Python conventions - for me it looked exactly like passing around `null` in Java.